packages/api/game/game.go

```diff
@@ -0,0 +1,253 @@
+package game
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/dopedao/dope-monorepo/packages/api/base"
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type Game struct {
+	Players      PlayersContainer
+	ItemEntities ItemEntitiesContainer
+
+	Register   chan *Player
+	Unregister chan *Player
+	Broadcast  chan BaseMessage
+}
+
+func (g *Game) Start(ctx context.Context) {
+	_, log := base.LogFor(ctx)
+
+	for {
+		select {
+		case player := <-g.Register:
+			g.Players.mutex.Lock()
```

Comment:
I would just use a single lock on the game object. You can embed it like

```
type Game struct {
   sync.Mutex
   players []*Player
}
```

and call it with `g.Lock()`

then remove `PlayersContainer` and just have a slice of players on `Game`